dipy/reconst/sfm.py

```diff
@@ -458,21 +475,45 @@ def fit(self, data, mask=None):
         isotropic = self.isotropic(self.gtab).fit(data, mask)
         flat_params = np.zeros((data_in_mask.shape[0],
                                 self.design_matrix.shape[-1]))
+        del data_in_mask
+        gc.collect()
+
         isopredict = isotropic.predict()
         if mask is None:
             isopredict = np.reshape(isopredict, (-1, isopredict.shape[-1]))
         else:
             isopredict = isopredict[mask]
 
-        for vox, vox_data in enumerate(flat_S):
-            # In voxels in which S0 is 0, we just want to keep the
-            # parameters at all-zeros, and avoid nasty sklearn errors:
-            if not (np.any(~np.isfinite(vox_data)) or np.all(vox_data == 0)):
-                fit_it = vox_data - isopredict[vox]
-                with warnings.catch_warnings():
-                    warnings.simplefilter("ignore")
-                    flat_params[vox] = self.solver.fit(self.design_matrix,
-                                                       fit_it).coef_
+        if n_threads > 1:
+            from collections import OrderedDict
+            try:
+                from joblib import Parallel, delayed
+            except ImportError as e:
+                print(e)
+
+            with Parallel(n_jobs=n_threads,
+                          backend='multiprocessing') as parallel:
+                out = parallel(
+                    delayed(self.fit_solver2voxels)(isopredict,
+                                                           vox_data, vox,
```

Comment:
deindent for match with line before: 
```suggestion
                                                    vox_data, vox,
```